packages/core-modules/contracts/modules/CoreElectionModule.sol

```diff
@@ -131,26 +136,55 @@ contract CoreElectionModule is IElectionModule, ElectionStorage, OwnableMixin {
         _electionStore().nextNominationPeriodPercent = percent;
     }
 
-    function elect(address[] memory candidates) external view override {
-        uint seatCount = _electionStore().nextSeatCount;
+    function elect(address[] memory candidates) external override {
+        ElectionStore storage store = _electionStore();
+        ElectionData storage electionData = store.electionData;
 
-        if (candidates.length != seatCount) {
+        if (candidates.length == 0) {
```

Comment:
I think we should also enforce a max number of candidates that can be voted for. Not just for algorithmic reasons, but to ensure the loop further down does not run out of gas.

Reply:
We can limit the number of inputs, but we cannot use seatCount for that, we can limit to the number of candidates (that is going to be finite). But limiting to the number of seats could make the election unresolved.

The simple case we discusses in our call (counting with preferences) was for 2 seats and 4 candidates and was resolved using the 3rd option.